incubator/hnc/internal/reconcilers/hnc_config.go

```diff
@@ -469,20 +469,26 @@ func (r *ConfigReconciler) loadNamespaceConditions(inst *api.HNCConfiguration) {
 			if len(nsnms) == 0 {
 				continue
 			}
+			// Sort namespaces and only set the first 3 namespaces in the condition if
+			// there are more than 3.
 			sort.Strings(nsnms)
-			status = append(status, api.ConditionAndAffectedNamespaces{
-				Type:       tp,
-				Reason:     reason,
-				Namespaces: nsnms,
-			})
+			l := len(nsnms)
+			// Message for 2 or 3 affected namespaces, e.g.
+			// 2 namespaces "d", "e" are affected by "ParentMissing"
+			msg := fmt.Sprintf("%v namespaces \"%s\" are affected by %q", l, strings.Join(nsnms, "\", \""), reason)
```

Comment:
I avoid using `%v` in a user-facing message - use a specific one like `%d` instead